omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -463,8 +463,15 @@ public static void setValue(Obs obs, Object value) throws ParseException, Conver
 							        + ", ConceptService.getFalseConcept expected");
 						}
 					}
+				}				
+				try {
+				     obs.setValueAsString(value.toString());
+				}catch(ParseException e) {
+					 Date dateWithTimeZone = (Date)ConversionUtil.convert(value.toString(), Date.class);
```

Comment:
What happens if this was not a date or datetime data type?

Reply:
For instance what happens for a numeric concept?

Reply:
Now theres already checks for different Obs.concept dataypes. 

Unless youre talking about a situation for example 
> where the the user sets a **Date concept datatype** for the obs , but then puts in a wrong **date value** may be like "6" ?

Reply:
Otherwise there are aready different checks for different Obs.ConceptDatatypes 
and this 
```
try {
				     obs.setValueAsString(value.toString());
	}catch(ParseException e) {
					 Date dateWithTimeZone = (Date)ConversionUtil.convert(value.toString(), Date.class);
 if(dateWithTimeZone != null) {
						 obs.setValueDatetime(dateWithTimeZone);
					 }
```
specificaly handles Datedatatypes

Reply:
All the other Concept datatypes are checked and handled above

Reply:
And actualy this method **setValueAsString(String s)** has  checks for almost all the different concept datatypes , so it handles each of them differently

Reply:
Are you sure that a numeric concept will not result into the execution of these changes?

Reply:
No , theres a check for a numeric concept ,way  before that lines of code 
`if (obs.getConcept().isNumeric()) { `  ,
 So a numeric concept is handled way before my changes

Reply:
Do you mean that a numeric concept will never execute your lines of code?

Reply:
Its strictly checked and handled  in here

				```
if (obs.getConcept().isNumeric()) {
					//get the actual persistent object rather than the hibernate proxy
					ConceptNumeric concept = Context.getConceptService().getConceptNumeric(obs.getConcept().getId());
					String units = concept.getUnits();
					if (StringUtils.isNotBlank(units)) {
						String originalValue = value.toString().trim();
						if (originalValue.endsWith(units))
							value = originalValue.substring(0, originalValue.indexOf(units)).trim();
						else {
							//check that that this value has no invalid units
							try {
								Double.parseDouble(originalValue);
							}
							catch (NumberFormatException e) {
								throw new APIException(originalValue + " has invalid units", e);
							}
						}
					}{
```

and that is far way before my lines of code. So theres no way it can be executed in my lines of code

Reply:
Do you get time and take a step back, to analyse your code in response to my comments? Are you sure that a concept numeric will never reach line 468?

Reply:
yes  understood what you were talking about . check out the correction